static/js/zamboni/devhub.js

```diff
@@ -97,7 +97,7 @@ $(document).ready(function() {
       var $isManualReview = $('#manual-review');
       var $submitAddonProgress = $('.submit-addon-progress');
       function updateListedStatus() {
-        if (!$isUnlisted.is(':checked') || $new_form.data('addon-is-listed')) {  // It's a listed add-on.
+        if (($isUnlisted.length && !$isUnlisted.is(':checked')) || $new_form.data('addon-is-listed')) {  // It's a listed add-on.
```

Comment:
Nope, it should be one or the other. The `$new_form.data('addon-is-listed')` is there (it's from the template/context) for uploads on existing add-ons, and the checkbox is there for uploads on a newly submitted add-on.